src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeFetchers.scala

```diff
@@ -69,7 +69,8 @@ trait BlocksDataFetchers {
     private def makeUrl = (offset: Offset) => s"blocks/${hashRef.value}~${String.valueOf(offset)}"
 
     //fetch a future stream of values
-    override val fetchData =
+    override val fetchData = {
+      logger.info("Fetching blocks")
```

Comment:
The logging here should be in the Kleisli constructor's internal function.
i.e. after the `offsets =>` are passed.
Calling `fetchData`is simply returning the kleisli wrapper around the real function. The kleisli returned is yet to be "run" at this point.

Reply:
Would you mind giving me the whole snippet for `fetchData()`? I am not familiar enough with the syntax to confidently make the change.